docs/getting_started/basic_usage/recurrent_subnet.rst

```diff
@@ -0,0 +1,54 @@
+.. _recurrent_subnet:
+
+======================
+Recurrent Sub-Networks
+======================
+
+For many task it will be necessary to define multiple that are applied as recurrent network over a sequential input,
+especially when running a search over sequences. While basic recurrent layers such as LSTM variants are defined by using
+the "``rec``" layer and selecting the desired "``unit``", custom sub-networks can be defined by passing a network
+dictionary for the "``unit``" attribute.  The defined structure will then be applied for each position of the sequence.
+As for the global network, an "``output``" layer is required to define which values will be the output of the subnet.
+The layer outputs of the previous timesteps can be accessed by adding the prefix "prev:" to the layer names. Static data
+from outside the subnet can be accessed via the layer prefix "base:".
```

Comment:
Some example units should be listed (here, or maybe below), e.g. `"NativeLstm2"` etc. And a link to the LSTM benchmark, such that the user can about relevant LSTM units.

Reply:
Or maybe not here at all, but somewhere else.